### ISSUE TITLE
Revision  of the Table of content

### ISSUE THREAD

**Issue by FabioPeruzzini** [FabioPeruzzini/actn-poi#6]
Revision of the table of content, taking into account a limited number of use cases with specific scenarios.
The intention is to avoid listing too many use cases to avoid being blocked by the IESG

**Comment by FabioPeruzzini** [comment]
This is the list of use cases I and Jeff think should be addressed by the draft:

Use Case 1 - Topology discovery, network inventory and multilayer correlation
1.1	Discovery of existing Och, ODU, IP links, IP tunnels and IP services

Use Case 2 – Provisioning of an IP Link over DWDM
 2.1 Provisioning of an IP link/LAG over DWDM with path latency constraint                                           
 2.2 Provisioning of an IP link/LAG over DWDM with path diversity constraint
                                          
Use Case 3 – Protection
 3.1 IP and DWDM protection coordination – Network resiliency during maintenance events
 3.2 Router port failure – 1:N “protection”

**Comment by danielkinguk** [comment]
Maybe we could combine Use Case 1 & 2, to reduce the perceived number of Use Cases. 

Use Case 1 - Service Provisioning of an IP Link over a Multi-layer DWDM Network
1.1 Topology Discovery, Network Inventory, and Multi-Layer Correlation (DWDM, Och, ODU, IP links, IP tunnels and IP Services)
1.2 Provisioning of an IP Link over the DWDM
1.3 Provisioning of an IP link/LAG over DWDM with using a Path Latency Constraint
1.4 Provisioning of an IP link/LAG over DWDM with using a Path Diversity Constraint

Use Case 2 – Service Protection over a Multi-layer DWDM Network
2.1 Protection Coordination of IP over DWDM Services
2.2 Ensuring Network Resiliency during Maintenance Events
2.2 Recovering from a Router Port Failure – 1:N “Protection”

If you agree, I will update the I-D based on something along the lines of the above.

**Comment by FabioPeruzzini** [comment]
I agree with the new structure (only two use cases).

In the use case 1, following Jeff's suggestion, I propose to add a new item. 
The new item is "Provioning of an additional link member to a LAG with phisical path constraint" (e.g. new optical connection need to follow the same physical path as the existing one).

About use case 2.1, as discussed with Micheal, I think it is better to skip it for the moment because it seems very complex to implement against very limited benefits.

Therefore, I propose the following table of contents:

Use Case 1 - Service Provisioning of an IP Link over a Multi-layer DWDM Network
1.1 Topology Discovery, Network Inventory, and Multi-Layer Correlation (DWDM, Och, ODU, IP links, IP tunnels and IP Services)
1.2 Provisioning of an IP Link/LAG over the DWDM
1.3 Provisioning of an IP link/LAG over DWDM with using a Path Latency Constraint
1.4 Provisioning of an IP link/LAG over DWDM with using a Path Diversity Constraint
1.5 Provisioning of an additional link member to a LAG with phisical path constraint

Use Case 2 – Service Protection over a Multi-layer DWDM Network
2.1 Ensuring Network Resiliency during Maintenance Events
2.2 Recovering from a Router Port Failure – 1:N “Protection”

**Comment by italobusi** [comment]
What about merging 1.3 and 1.4 as "Provisioning of an IP link/LAG over DWDM with using a Path Constraints"?

In principle, any constraints can be requested for path setup so we can make the description more generic, clarifying that Latency and Diversity are the most important constraints used in real network deployments

BTW, would it be possible to request constraints on both Latency and Diversity at the same time?

**Comment by michael-scharf** [comment]
I am aware of a lot of business value just by the discovery use case only. In think there are network operators who only care about discovery. In their use case, there is no need for any automated provisioning action. In other words, the reason for topology discovery is NOT to provision an IP link.
The typical example is IP/optical alarm correlation. And this is is not the only use case for which only discovery matters.

BTW, discovery can be useful even if the DWDM network is relatively static and does not allow a lot of automation in the topical network (e.g., due to lack of ROADMs). In that case, the transport network can hardly automate IP link provisioning. Still, the IP/optical topology correlation may be very useful for network operation.

Personally, I found the original use case list starting with case 1 "discovery" much closer to what operators may deploy, also from a timing perspective. Many operators who currently run entirely isolated IP and transport networks would probably start with a "discovery" use case as a first step, e.g., within 1-2 years from now. Automating IP link/LAG provisioning may sound like research to such an operator, i.e., it may be 3-5 years in the future for them.

I think many operators would just deploy a discovery solution as a first step in a production network, and then possibly grow from there. Having a dedicated use case 1 for the first step would reflect that. But, well, I am an academic now and I can't speak for operators ;-)

**Comment by italobusi** [comment]
@FabioPeruzzini @michael-scharf @danielkinguk : I think this issue has been already closed with the -02 version of the draft:

https://tools.ietf.org/html/draft-peru-teas-actn-poi-applicability-02

Can we close it?

What do you think?

**Comment by italobusi** [comment]
**2020-04-27 Call**

It is worthwhile reviewing the ToC after the merge with draft-lee and the draft can put more emphasis on using POI to support L2/L3 services

**Comment by jbouqui153** [comment]
As agreed in our last call I leave here a draft proposal for a new table of content with the aim in mind to integrate better the service part (V03 section 5) and the transport part (Section 1, 2, 3 and 4). In the document attached I put the V03 table of content as reference only and then you have the new proposal which explains what would be kept unchanged from V03 and mapped to which of the new sections. Fabio told me he is fine with the proposal. Just want to make clear that all the content of V03 is maintained but relocated throughout the new sections proposed. Use cases (Scenarios) are modified following more the draft-lee (being service focused) however many existing content of the transport scenarios of V03 will be reused as well. All your comments are welcome.
[POI draft - New Table of Content Draft Proposal v2.docx](https://github.com/FabioPeruzzini/actn-poi/files/4599759/POI.draft.-.New.Table.of.Content.Draft.Proposal.v2.docx)
.

**Comment by italobusi** [comment]
> As agreed in our last call I leave here a draft proposal for a new table of content with the aim in mind to integrate better the service part (V03 section 5) and the transport part (Section 1, 2, 3 and 4). In the document attached I put the V03 table of content as reference only and then you have the new proposal which explains what would be kept unchanged from V03 and mapped to which of the new sections. Fabio told me he is fine with the proposal. Just want to make clear that all the content of V03 is maintained but relocated throughout the new sections proposed. Use cases (Scenarios) are modified following more the draft-lee (being service focused) however many existing content of the transport scenarios of V03 will be reused as well. All your comments are welcome.
> [POI draft - New Table of Content Draft Proposal v2.docx](https://github.com/FabioPeruzzini/actn-poi/files/4599759/POI.draft.-.New.Table.of.Content.Draft.Proposal.v2.docx)
> .

The proposal looks a good starting point

Few comments for discussion:

1) I think we may not split between sections 2 and 3: sections 3.x can become 2.x

2) What about the scenario "Provisioning Link Members to an existing LAG" (section 3.4 of the -03 draft)?
Can we add this scenario within new section 4.2?

3) What about the scenario "Router Port Failure" (section 4.2 of the -03 draft)?
Can we split section 4 as in -03 draft between one section for service and one section for resilience?

**Comment by sergiobelotti** [comment]
> As agreed in our last call I leave here a draft proposal for a new table of content with the aim in mind to integrate better the service part (V03 section 5) and the transport part (Section 1, 2, 3 and 4). In the document attached I put the V03 table of content as reference only and then you have the new proposal which explains what would be kept unchanged from V03 and mapped to which of the new sections. Fabio told me he is fine with the proposal. Just want to make clear that all the content of V03 is maintained but relocated throughout the new sections proposed. Use cases (Scenarios) are modified following more the draft-lee (being service focused) however many existing content of the transport scenarios of V03 will be reused as well. All your comments are welcome.
> [POI draft - New Table of Content Draft Proposal v2.docx](https://github.com/FabioPeruzzini/actn-poi/files/4599759/POI.draft.-.New.Table.of.Content.Draft.Proposal.v2.docx)
> .

I think it is a very good proposal of restructuring content to host and harmonise the text coming for the former POI-ACTN draft.
I like take separate section 2 , just explicit for Reference architecture and netowrk scenario and  the rest the document.
About the reference scenario I'd like to put on the table if it is foreseen as future reference scenario to deal with also L1 topology , with OTN switch other than just ROADM .
In the section 3.3 it is mentioned details regarding Service to TE tunnel mapping for hard or soft isolation. Is it foreseen to add something regarding "slicing" here ? I ask since in IETF there is a DT for Transport Slicing and in that context a lot of debate has been done on the role of isolation .

**Comment by italobusi** [comment]
**2020-05-11 Call**

Agree ToC:

1.	Introduction
2.	Reference architecture and network scenario
2.1. L2/L3VPN/VN Service Request by the Customer
2.2. Service and Network Orchestration
2.3. IP/MPLS Domain Controller and NE Functions
2.4. Optical Domain Controller and NE Functions
3.	Multi-layer and multi-domain services scenarios
3.1 Scenario 1: network and service topology discovery
3.2 Scenario 2: L3VPN/L2VPN establishment
4.	Security considerations
5.	Operational considerations
Appendix A - Multi-layer and multi-domain resiliency
A.1 Maintenance Window
A.2 Router port failure

**Comment by ggrammel** [comment]
What exactly is meant by "L2/L3VPN/VN Service Request by the Customer"? The issues I have with the headline are:

1. I can set-up a L3VPN from home without asking my carrier for permission, worst case some firewall entries in my home gateway need to be tweaked. So most likely 2.1 is meant as Carrier administered VPN.  "Service request by the customer" is at least misleading in this case. If it is a Carrier service it should be spelled out. how about renaming it "L2/L3VPN/VN carrier service" 
2. An observation: L2 services (EPL, EVPL) are usually provided over the Internet, so an EPL/EVPL becomes in effect a L3 service. That doesn't fit very well in a layered model. If an L2 service is supposed to be limited to native Ethernet switched services over transport (excluding IP/MPLS) then we better take a note going forward.
3. Services provided over the Internet do not necessarily have a bandwidth associated. Are those meant to be included in the term "L2/L3VPN/VN " or is some bandwidth guarantee implicitly implied? If that's the case we may want to call it "L2/L3VPN/VN carrier service with bandwidth guarantee" 
4. Another observation: L3VPN services over the internet can have end-points in different networks. By limiting the scope to carrrier provisioned VPN one could argue that both ends stay within an administrative domain. 

in summary: if the intention is to cover all VPN/VN services defined by IETF and break them down in different use cases 2.1 should probably renamed "L1/L2/L3 VPN services". Else, if the scope is more limited e.g. covering carrier provisioning and bandwidth guarantee, it would make sense to indicate it directly in 2.1 e.g. something like "L2/L3VPN/VN carrier service with bandwidth guarantee to not create false associations.

**Comment by michael-scharf** [comment]
I am also confused by the focus on "L2/L3VPN/VN Service Request by the Customer".

I agree with Gert that in particular L3VPNs seem like an open-ended area. For instance, L3VPNs with Inter-AS Options A, B, C can span multiple carriers. Traffic engineering across several AS is an interesting problem, but by and large orthogonal to optical DWDM systems - at least as far as I understand. Also, keep in mind that a L3VPN may run over a L2VPN and there is not a single "customer" in this case. It is a complex layered system...

And I don't understand how the ToC reflects the following (simpler) bandwidth-on-demand scenario: A carrier observes that the bandwith of an MPLS LSP has to increase, e.g., by adding an IP link or growing a LAG. He needs a new lambda in the DWDM layer for that. How does this use case fit into the ToC? Is this not POI?

And, well, I believe that increasing the bandwidth of an MPLS LSP or an IP link used by MPLS/IP can be required without any IP-based L3VPN/L2VPN "customer" service request. For instance, this can just be required if the routing in MPLS/IP changes as part of the normal network operation, e.g., to optimize the IP traffic distribution (or maintenance and the like).

At least in earlier days the use cases listed at https://sedonasys.com/ were sort of a reference to those in industry dealing with POI. It is a bit difficult to map this list of use cases to the proposed ToC.

**Comment by jbouqui153** [comment]
Please refer to new version just uploaded where section 2 and section 2.2 has been modified to highlight that OSS/Orchestration layer is expected in North Bound of the MDSC (and not a CNC) and there are two types of use cases: either L2/L3VPN/VN requested to MDSC by OSS/Orchestration layer or initiated by MDSC itself, like discovery, multi-layer/multi-domain optimizations and/or maintenance works (e.g. rerouting LSPs with their associated services when putting a fibre in maintenance mode during a maintenance window). This aims to remove the lack of clarity of previous version mentioning L2/L3VPN/VN requested by customer (old title of section 2.2).

**Comment by Paolo-Volpato** [comment]
This is the contribution on the status of orchestrators/controllers in the industry, as discussed during the call of Monday July 27. As noted:
- the MDSC NBI is probably out of the scope of ACTN (exception is the NBI for slicing), and
- the MPI between the Service and the Network component of MDSC has to be further clarified/investigated.
[Comparison_short_v3.pptx](https://github.com/FabioPeruzzini/actn-poi/files/4982410/Comparison_short_v3.pptx)
